src/monitoring_frame_msg.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2020 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#include <algorithm>
+#include <cstdint>
+#include <iterator>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "psen_scan_v2/monitoring_frame_format_error.h"
+#include "psen_scan_v2/monitoring_frame_msg.h"
+#include "psen_scan_v2/raw_processing.h"
+#include "psen_scan_v2/raw_scanner_data.h"
+
+namespace psen_scan_v2
+{
+constexpr FieldHeader::Id AdditionalFieldIds::SCAN_COUNTER;
+constexpr FieldHeader::Id AdditionalFieldIds::MEASURES;
+constexpr FieldHeader::Id AdditionalFieldIds::END_OF_FRAME;
+
+FieldHeader::FieldHeader(std::istringstream& is)
+{
+  raw_processing::read(is, id_);
+  raw_processing::read(is, length_);
+  length_--;
+}
+
+MonitoringFrameMsg MonitoringFrameMsg::fromRawData(const MaxSizeRawData& data)
```

Comment:
I think this function could be much simple by relying on `raw_processing::read` and maybe some variants of it.
`deserializeAdditionalField` could/should be simplified by a simple loop + switch case. As long as we are doing simple deserialization there is no benefit is adding a strategy pattern. I'd rather argument that this is more error prone due to the complexity of passing a changing stream around...